src/widgets.jl

```diff
@@ -328,6 +328,59 @@ end
 button(label::Union{String,Symbol}; widget=nothing, observable=nothing, own=nothing) =
     button(; label=label, widget=widget, observable=observable, own=own)
 
+######################### ColorButton ###########################
+
+struct ColorButton{C} <: InputWidget{Nothing}
+    observable::Observable{C}
+    widget::GtkColorButtonLeaf
+
+    function ColorButton{C}(observable::Observable{C}, widget) where C <: Colorant
+        obj = new(observable, widget)
+        gc_preserve(widget, obj)
+        obj
+    end
+end
+
+colorbutton(observable::Observable{C}, widget::GtkColorButtonLeaf) where C <: Colorant =
+    ColorButton{C}(observable, widget)
+
+"""
+    colorbutton(color; widget=nothing, observable=nothing)
+    colorbutton(; color=nothing, widget=nothing, observable=nothing)
+
+Create a push button with color `color`. Clicking opens a color picker. Optionally provide:
+  - a GtkButton `widget` (by default, creates a new one)
+  - the (Observables.jl) `observable` coupled to this button (by default, creates a new observable)
+"""
+function colorbutton(;
+                color::C = RGBA(0, 0, 0, 0),
+                widget=nothing,
+                observable=nothing,
+                own=nothing) where C <: Colorant
+    obsin = observable
+    if observable === nothing
+        observable = Observable(color)
+    end
+    if own === nothing
+        own = observable != obsin
+    end
+    gcolor = Gtk.GdkRGBA(color.r, color.g, color.b, color.alpha)
```

Comment:
You're not enforcing that `C` is of type `RGBA` and so you can't guarantee that `color` has these fields. You could use `C<:AbstractRGBA`. Test using `RGBA32` and you'll see that this fails because that type doesn't have these fields either. It's best to use `red(c)`, `green(c)`, etc instead.

Alternatively you can use `convert(RGBA, c)` and support arbitrary color spaces.

Reply:
After thinking about it, I think supporting `Color{T, 3}` makes the most sense. The color picker doesn't support transparency. The Observable can be whatever concrete color type the user wants otherwise. Internally, it'll convert from GdkRGBA to RGBA, which is converted to the Observable concrete type. And otherwise use the red/green/blue methods to convert from the concrete type to GdkRGBA.